e2e: TestAPMAssociationWhenReferencedESDisappears fails with stack 7.3

Error:
```
17:03:16      --- FAIL: TestAPMAssociationWhenReferencedESDisappears/Events_should_eventually_show_up_in_Elasticsearch (300.00s)
17:03:16          require.go:794: 
17:03:16              	Error Trace:	utils.go:80
17:03:16              	Error:      	Received unexpected error:
17:03:16              	            	404 Not Found: no such index [apm-7.3.0-metric-2017.05.30]
17:03:16              	Test:       	TestAPMAssociationWhenReferencedESDisappears/Events_should_eventually_show_up_in_Elasticsearch
```
https://devops-ci.elastic.co/job/cloud-on-k8s-e2e-tests/247/
https://devops-ci.elastic.co/job/cloud-on-k8s-e2e-tests/248/

--- a/operators/Makefile
+++ b/operators/Makefile
@@ -320,6 +320,7 @@ e2e-local:
 	@go run test/e2e/cmd/main.go run \
 		--test-run-name=e2e \
 		--test-context-out=$(LOCAL_E2E_CTX) \
+		--elastic-stack-version=$(STACK_VERSION) \
 		--auto-port-forwarding \
 		--local
 	@test/e2e/run.sh -run "$(TESTS_MATCH)" -args -testContextPath $(LOCAL_E2E_CTX)

--- a/operators/test/e2e/test/apmserver/builder.go
+++ b/operators/test/e2e/test/apmserver/builder.go
@@ -30,6 +30,11 @@ func NewBuilder(name string) Builder {
 			Spec: apmtype.ApmServerSpec{
 				NodeCount: 1,
 				Version:   test.Ctx().ElasticStackVersion,
+				Config: &commonv1alpha1.Config{
+					Data: map[string]interface{}{
+						"apm-server.ilm.enabled": false,
+					},
+				},
 				PodTemplate: corev1.PodTemplateSpec{
 					Spec: corev1.PodSpec{
 						SecurityContext: test.DefaultSecurityContext(),

